### PR TITLE
[WT-579] Visit source contacts

### PR DESCRIPTION
**Pull request by Deacon-Coding**
### Description
added visitSource "How did you hear about us?" to contact.service and contact.router 

### Risks
none known

### Validation
changes appear in localhost

### Issue
https://github.com/WonderTix/WonderTix/issues/579

### Operating System
Windows 11

### CODE REVIEW

**Comment by ben16williams** [comment]
I believe you'll need to add a `visitsource` string field to the `contacts` table for values to actually be saved. [Here's ](https://github.com/WonderTix/WonderTix/blob/main/server/prisma/schema.prisma#L11) the contact table in the schema that would updated and [here's ](https://github.com/WonderTix/WonderTix/pull/543#issuecomment-1873003064) a comment on how to do a DB migration with Prisma

File: server/src/api/contacts/contacts.service.ts
Comment (benfaustin95) [inline server/src/api/contacts/contacts.service.ts:1]: We moved away from api v1 for contact information, I think this change might need to be made in `contactController.ts` and `eventController.service.ts` to have effect.

**Comment by benfaustin95** [comment]
It looks like visitsource is not making its way into the database. In order to save visit source with the contact it needs to be added to the `validateContact` function `eventController.service.ts`.

File: server/src/controllers/eventController.service.ts [hunk 1]
```diff
@@ -311,6 +311,7 @@ const validateContact = (formData: checkoutForm) => {
           new RegExp('^(\\+?\\d{1,2}\\s?)?\\(?\\d{3}\\)?[\\s.-]?\\d{3}[\\s.-]?\\d{4}$'),
       ),
     }),
+    ...(formData.visitSource && {visitSource: formData.visitSource}),
```
Comment (ben16williams) [inline server/src/controllers/eventController.service.ts:314]: The new column name in `contacts` table is `visitsource` with a lowercase 's'. Updating this line to `...(formData.visitSource && {visitsource: formData.visitSource}),` should get things working

**Comment by benfaustin95** [comment]
I think this just needs to be merged with the most recent updates to main and it should be good to go.

**Review by benfaustin95 — DISMISSED** [review]

**Review by henrykaus — COMMENTED** [review]
The v1 APIs were just removed as they aren't used anywhere. So all changes should occur in only the Prisma (v2) APIs.

**Review by henrykaus — APPROVED** [review]
Looks good!

**Pull request merged**